aospy/test/test_automate.py

```diff
@@ -0,0 +1,187 @@
+import builtins
+import sys
+
+import pytest
+
+from aospy import Var, Proj
+from aospy.automate import (_get_attr_by_tag, _permuted_dicts_of_specs,
+                            _get_all_objs_of_type, _merge_dicts,
+                            _input_func_py2_py3, AospyException,
+                            _user_verify, CalcSuite)
+from .data.objects import examples as lib
+from .data.objects.examples import (
+    example_proj, example_model, example_run, condensation_rain,
+    convection_rain, precip, ps, sphum, globe, sahel
+)
+
+
+@pytest.fixture
+def obj_lib():
+    return lib
+
+
+@pytest.fixture
+def all_vars():
+    return {condensation_rain, convection_rain, precip, ps, sphum}
+
+
+@pytest.fixture
+def all_projects():
+    return {example_proj}
+
+
+@pytest.fixture
+def all_models():
+    return {example_model}
+
+
+@pytest.fixture
+def all_runs():
+    return {example_run}
+
+
+@pytest.fixture
+def all_regions():
+    return {globe, sahel}
+
+
+@pytest.fixture
+def calcsuite_specs():
+    return {
+        'time_offset': [None],
+        'date_range': ['default'],
+        'intvl_in': ['monthly'],
+        'region': [(globe, sahel)],
```

Comment:
Yes this is a good point, and the same holds for `output_time_regional_reductions`.  The backend should handle this, rather than forcing the user to.